library/types/test/string_test.rb

```diff
@@ -0,0 +1,176 @@
+#!/usr/bin/env rspec
+
+require_relative "test_helper"
+
+Yast.import "String"
+
+describe Yast::String do
+  before do
+    # ensure proper default locale
+    ENV["LANG"] = "C"
+  end
+
+  describe ".Quote" do
+    it "returns empty string if nil passed" do
+      expect(subject.Quote(nil)).to eq ""
+    end
+
+    it "returns all single quotes escaped" do
+      expect(subject.Quote("")).to eq ""
+      expect(subject.Quote("a")).to eq "a"
+      expect(subject.Quote("a'b")).to eq "a'\\''b"
+      expect(subject.Quote("a'b'c")).to eq "a'\\''b'\\''c"
+    end
+  end
+
+  describe ".UnQuote" do
+    it "returns empty string if nil passed" do
+      expect(subject.UnQuote(nil)).to eq ""
+    end
+
+    it "returns unescaped single quotes" do
+      expect(subject.UnQuote("")).to eq ""
+      expect(subject.UnQuote("a")).to eq "a"
+      expect(subject.UnQuote("a'\\''b")).to eq "a'b"
+      expect(subject.UnQuote("a'\\''b'\\''c")).to eq "a'b'c"
+      expect(subject.UnQuote("a'\\'''\\''b'\\''c")).to eq "a''b'c"
+    end
+  end
+
+  describe ".FormatSize" do
+    it "returns empty string if nil passed" do
+      expect(subject.FormatSize(nil)).to eq ""
+    end
+
+    FORMAT_SIZE_DATA = {
+      0                     => "0 B",
+      1                     => "1 B",
+      1025                  => "1 KiB",
+      1125                  => "1.1 KiB",
+      743 * 1024            => "743 KiB",
+      1_049_000             => "1.00 MiB",
+      -1_049_000            => "-1 MiB", # FIXME: why?
+      1_074_000_000         => "1.000 GiB",
+      1_100_000_000_000     => "1.000 TiB",
+      1_126_000_000_000_000 => "1024.091 TiB",
+      1 << 10               => "1 KiB",
+      1 << 20               => "1.00 MiB",
+      1 << 30               => "1.000 GiB",
+      1 << 40               => "1.000 TiB"
+    }
+
+    it "returns size formatted with proper bytes units" do
+      FORMAT_SIZE_DATA.each do |arg, res|
+        expect(subject.FormatSize(arg)).to eq res
+      end
+    end
+  end
+
+  describe ".FormatSizeWithPrecision" do
+    it "returns empty string if nil passed" do
+      expect(subject.FormatSizeWithPrecision(nil, nil, nil)).to eq ""
+    end
+
+    it "returns bytes in proper unit with passed precision forcing trailing zeroes if omit_zeroes not passed" do
```

Comment:
This description is quite hard to read and understand. Maybe breaking it in two or even using contexts (to group omit_zeroes true/false, for example) would help.